src/main/scala/org/neo4j/spark/reader/Neo4jInputPartitionReader.scala

```diff
@@ -15,27 +15,37 @@ import scala.collection.JavaConverters._
 class Neo4jInputPartitionReader(private val options: Neo4jOptions,
                                 private val filters: Array[Filter],
                                 private val schema: StructType,
-                                private val jobId: String) extends InputPartition[InternalRow]
+                                private val jobId: String,
+                                private val partitionId: Int = 0,
+                                private val skip: Long = -1L,
+                                private val limit: Long = -1L) extends InputPartition[InternalRow]
   with InputPartitionReader[InternalRow]
   with Logging {
 
   private var result: Iterator[Record] = _
   private var session: Session = _
   private var transaction: Transaction = _
-  private val driverCache: DriverCache = new DriverCache(options.connection, jobId)
+  private val driverCache: DriverCache = new DriverCache(options.connection,
+    if (partitionId > 0) s"$jobId-$partitionId" else jobId)
 
-  private val query: String = new Neo4jQueryService(options, new Neo4jQueryReadStrategy(filters)).createQuery()
+  private val query: String = new Neo4jQueryService(options, new Neo4jQueryReadStrategy(filters))
+    .createQuery()
+    .concat(if (skip != -1 && limit != -1) s" SKIP $$skip LIMIT $$limit" else "")
```

Comment:
it would be nicer if we had a proper method on createQuery i.e. pass it to the query builder

Reply:
Done